core/src/test/java/io/github/tcdl/msb/message/MessageFactoryTest.java

```diff
@@ -219,4 +220,32 @@ public void testCreateAckBuilder() throws Exception {
         Acknowledge ack = messageFactory.createAckBuilder().build();
         assertNotNull(ack.getResponderId());
     }
+
+    @Test
+    public void testCreateRequestMessageBuilderPublishedAtPresent() {
+        String bodyText = "body text";
+        Payload requestPayload = TestUtils.createPayloadWithTextBody(bodyText);
+
+        Builder requestMessageBuilder = TestUtils.createMessageBuilder(FIXED_CLOCK);
+
+        Message message = messageFactory.createRequestMessage(requestMessageBuilder, requestPayload);
+
+        assertNotNull(message.getMeta().getPublishedAt());
+        assertEquals(message.getMeta().getPublishedAt(), FIXED_CLOCK.instant());
+    }
+
+    @Test
+    public void testCreateRequestMessageBuilderPublishedAtIsAfterCreatedAt() {
+        String bodyText = "body text";
+        Payload requestPayload = TestUtils.createPayloadWithTextBody(bodyText);
+
+        Builder requestMessageBuilder = TestUtils.createMessageBuilder(null);
```

Comment:
I think it would be more expressive if you pass here `Clock.systemDefaultZone()` instead of `null`